pkg/featuregate/featuregate.go

```diff
@@ -29,6 +29,13 @@ var (
 		"operator.autoinstrumentation.dotnet",
 		featuregate.StageBeta,
 		featuregate.WithRegisterDescription("controls whether the operator supports .NET auto-instrumentation"))
+
+	// EnableTargetAllocatorRewrite is the feature gate that controls whether the collector's configuration should
+	// automatically be rewritten when the target allocator is enabled.
+	EnableTargetAllocatorRewrite = featuregate.GlobalRegistry().MustRegister(
+		"operator.enableTargetAllocatorRewrite",
```

Comment:
Sticking with the hierarchical pattern, how do you feel about `operator.targetallocator.rewrite`?

Reply:
That makes it sound like this changes target allocator behaviour, but in reality we're changing how the operator rewrites prometheusreceiver config. So maybe something like `operator.prometheusreceiver.rewritetargetallocator`? Kind of a mouthful, but more descriptive imo.

Reply:
Since the flag ultimately affects the collector how about `operator.collector.rewritetargetallocator`.  I think that would scale well with any other flags that mess with collector configuration/capabilities.

Reply:
Yeah, that sounds better.

Reply:
Renamed.